hermes-console-vue/src/views/admin/constraints/create-constraint-form/CreateConstraintFormView.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { describe, expect } from 'vitest';
+import { render, renderWithEmits } from '@/utils/test-utils';
+import { waitFor } from '@testing-library/vue';
+import CreateConstraintFormView from '@/views/admin/constraints/create-constraint-form/CreateConstraintFormView.vue';
+
+describe('CreateConstraintForm', () => {
+  it('should emit a cancel event when user clicks cancel button', async () => {
+    // given
+    const wrapper = renderWithEmits(CreateConstraintFormView, {
+      props: {
+        subscription: false,
+      },
+    });
+
+    await wrapper
+      .find('[data-testid="createConstraintCancel"]')
+      .trigger('click');
+
+    await waitFor(() => {
+      const cancelEvents = wrapper.emitted('cancel')!!;
+      expect(cancelEvents).toHaveLength(1);
+    });
+  });
+
+  it('should render subscription name field when in subscription mode', async () => {
+    // given
+    const { queryAllByText } = render(CreateConstraintFormView, {
+      props: {
+        subscription: true,
+      },
+    });
+    // when
+    expect(
+      queryAllByText('constraints.createForm.subscriptionName')[0],
+    ).toBeVisible();
+  });
+
+  it('should not render subscription name field when in topic mode', async () => {
+    // given
+    const { queryByText } = render(CreateConstraintFormView, {
+      props: {
```

Comment:
what do You think about defining 'const props' object in test class scope, and then using the same props in all test cases? It would reduce boilerplate